api/albums-router.js

```diff
@@ -7,35 +7,37 @@ const {
   joinAlbum,
   shareAlbum,
 } = require('./albums');
-const { addShareToken } = require('./user');
+const { addShareToken, getSharedUsers } = require('./user');
 
 const router = Router();
 
 router.post('/memberships', async (req, res) => {
   const { shareToken } = req.body;
   const userToken = req.user.token;
-  const userId = req.user.profile.id;
 
   console.log('who is user?', req.user);
 
   try {
-    const album = await joinAlbum(userToken, userId, shareToken);
+    const album = await joinAlbum(userToken, shareToken);
     res.json({ album });
   } catch (err) {
     console.log('ERRRRRRR', err);
     res.status(500).json(err);
   }
 });
 
-router.post('/:albumId/share-token', async (req, res) => {
+router.post('/:albumId/activate', async (req, res) => {
   const { albumId } = req.params;
   const userToken = req.user.token;
-  const userId = req.user.profile.id;
-  const shareUser = req.body.shareUser;
+
   try {
-    const album = await shareAlbum(userToken, userId, albumId, shareUser);
-    await addShareToken(shareUser);
-    res.json({ album });
+    // this will add sharetokens to users who were tagged in this specific album
+    const shareToken = await activateAlbum(userToken, albumId);
```

Comment:
Are we importing this function somewhere??

Reply:
it's actually incredible how many mistakes my IDE prevents...i think that's probably not a good thing lol